src/Common/OvercommitTracker.cpp

```diff
@@ -23,8 +23,16 @@ OvercommitTracker::OvercommitTracker(std::mutex & global_mutex_)
     , allow_release(true)
 {}
 
+#define LOG_DEBUG_SAFE(...)                       \
+    {                                             \
+        OvercommitTrackerBlockerInThread blocker; \
+        LOG_DEBUG(__VA_ARGS__);                   \
+    }
```

Comment:
You can slightly modify macro to preserve semicolons where the macro is used in the code:
```
#define LOG_DEBUG_SAFE(...)                       \
    do {                                          \
        OvercommitTrackerBlockerInThread blocker; \
        LOG_DEBUG(__VA_ARGS__);                   \
    } while (false)
```